gravitee-apim-console-webui/src/management/api/endpoints-v4/endpoint-group/create/api-endpoint-group-create.component.spec.ts

```diff
@@ -0,0 +1,369 @@
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { MatIconTestingModule } from '@angular/material/icon/testing';
+import { FormsModule } from '@angular/forms';
+import { InteractivityChecker } from '@angular/cdk/a11y';
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { MatDialogModule } from '@angular/material/dialog';
+import { MatStepHarness } from '@angular/material/stepper/testing';
+import { MatSnackBarModule } from '@angular/material/snack-bar';
+
+import { ApiEndpointGroupCreateComponent } from './api-endpoint-group-create.component';
+import { ApiEndpointGroupCreateHarness } from './api-endpoint-group-create.harness';
+
+import { UIRouterState, UIRouterStateParams } from '../../../../../ajs-upgraded-providers';
+import { ApiEndpointGroupModule } from '../api-endpoint-group.module';
+import { CONSTANTS_TESTING, GioHttpTestingModule } from '../../../../../shared/testing';
+import { ApiV4, EndpointGroupV4, EndpointV4Default, fakeApiV4 } from '../../../../../entities/management-api-v2';
+
+const API_ID = 'api-id';
+const ENDPOINT_GROUP_NAME = 'My Endpoint Group';
+const FAKE_UI_ROUTER = { go: jest.fn() };
+const fakeKafkaSchema = {
+  $schema: 'http://json-schema.org/draft-07/schema#',
+  type: 'object',
+  properties: {
+    topic: {
+      title: 'Kafka topic',
+      type: 'string',
+      description: 'A kafka topic',
+    },
+  },
+  additionalProperties: false,
+  required: ['topic'],
+};
+
+const fakeRabbitMqSchema = {
+  $schema: 'http://json-schema.org/draft-07/schema#',
+  type: 'object',
+  properties: {
+    rabbitFood: {
+      title: 'Rabbit food',
+      type: 'string',
+      description: 'Some rabbit food',
+    },
+  },
+  additionalProperties: false,
+  required: ['rabbitFood'],
+};
+
+const ENDPOINT_LIST = [
+  {
+    id: 'mock',
+    name: 'Mock',
+    description: 'Use a Mock backend to emulate the behaviour of a typical HTTP server and test processes',
+    icon: 'mock-icon',
+    deployed: true,
+    supportedApiType: 'MESSAGE',
+  },
+  {
+    id: 'rabbitmq',
+    name: 'RabbitMQ',
+    description: 'RabbitMQ Endpoint',
+    icon: 'rabbitmq-icon',
+    deployed: true,
+    supportedApiType: 'MESSAGE',
+  },
+  {
+    id: 'kafka',
+    name: 'Kafka',
+    description: 'Publish and subscribe to messages from one or more Kafka topics',
+    icon: 'kafka-icon',
+    deployed: true,
+    supportedApiType: 'MESSAGE',
+  },
+];
+describe('ApiEndpointGroupCreateComponent', () => {
+  let httpTestingController: HttpTestingController;
+  let fixture: ComponentFixture<ApiEndpointGroupCreateComponent>;
+  let harness: ApiEndpointGroupCreateHarness;
+  let api: ApiV4;
+  let routerSpy: any;
+
+  const initComponent = async (testApi: ApiV4) => {
+    const routerParams: unknown = { apiId: API_ID };
+
+    api = testApi;
+
+    TestBed.configureTestingModule({
+      imports: [
+        NoopAnimationsModule,
+        MatDialogModule,
+        MatSnackBarModule,
```

Comment:
Good question!

Reply:
They're actually not in the ApiEndpointGroupModule, I imagine because they are imported more globally? I have moved them into the Module and out of testing to make it clearer.